equinix_metal.tf

```diff
@@ -3,7 +3,7 @@ resource "tls_private_key" "ssh_key_pair" {
   rsa_bits  = 4096
 }
 
-resource "metal_ssh_key" "ssh_pub_key" {
+resource "equinix_ssh_key" "ssh_pub_key" {
```

Comment:
This should be [`equinix_metal_ssh_key`](https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_ssh_key)